modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManagerImpl.java

```diff
@@ -17,53 +17,282 @@
+import org.apache.ignite.metastorage.common.Conditions;
+import org.apache.ignite.metastorage.common.Key;
+import org.apache.ignite.metastorage.common.Operations;
+import org.apache.ignite.metastorage.common.WatchEvent;
+import org.apache.ignite.metastorage.common.WatchListener;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.raft.client.service.RaftGroupService;
+import org.apache.ignite.table.Table;
 import org.apache.ignite.table.manager.TableManager;
+import org.jetbrains.annotations.NotNull;
 
-/**
- * Table Manager that handles inner table lifecycle and provide corresponding API methods.
- */
-// TODO: IGNITE-14586 Remove @SuppressWarnings when implementation provided.
-@SuppressWarnings({"FieldCanBeLocal", "unused"}) public class TableManagerImpl implements TableManager {
-    /** Meta storage service. */
-    private final MetaStorageManager metaStorageMgr;
+public class TableManagerImpl implements TableManager {
+    /** The logger. */
+    private static final IgniteLogger LOG = IgniteLogger.forClass(TableManagerImpl.class);
 
-    /** Network cluster. */
-    private final ClusterService clusterNetSvc;
+    /** Internal prefix for the metasorage. */
+    private static final String INTERNAL_PREFIX = "internal.tables.";
 
-    /** Schema manager. */
-    private final SchemaManager schemaMgr;
+    /** Meta storage service. */
+    private final MetaStorageManager metaStorageMgr;
 
     /** Configuration manager. */
     private final ConfigurationManager configurationMgr;
 
-    /** Raft manager. */
-    private final Loza raftMgr;
+    /** Table creation subscription future. */
+    private CompletableFuture<Long> tableCreationSubscriptionFut;
+
+    /** Tables. */
+    private Map<String, Table> tables;
 
     /**
-     * The constructor.
-     *
-     * @param configurationMgr Configuration table.
-     * @param clusterNetSvc Cluster network service.
-     * @param metaStorageMgr MetaStorage manager.
-     * @param schemaMgr Schema manager.
+     * @param configurationMgr Configuration manager.
+     * @param metaStorageMgr Meta storage manager.
+     * @param schemaManager Schema manager.
      * @param raftMgr Raft manager.
      */
     public TableManagerImpl(
         ConfigurationManager configurationMgr,
-        ClusterService clusterNetSvc,
         MetaStorageManager metaStorageMgr,
-        SchemaManager schemaMgr,
+        SchemaManager schemaManager,
         Loza raftMgr
     ) {
+        tables = new HashMap<>();
+
         this.configurationMgr = configurationMgr;
-        this.clusterNetSvc = clusterNetSvc;
         this.metaStorageMgr = metaStorageMgr;
-        this.schemaMgr = schemaMgr;
-        this.raftMgr = raftMgr;
+
+        String localNodeName = configurationMgr.configurationRegistry().getConfiguration(NetworkConfiguration.KEY)
+            .name().value();
+
+        configurationMgr.configurationRegistry().getConfiguration(LocalConfiguration.KEY)
```

Comment:
I believe that we should listen ClusterConfiguration.metastorageClusterNodeNames instead of local ones.

Reply:
Done.